Source/Core/Core/Boot/DolReader.cpp

```diff
@@ -124,3 +140,80 @@ bool DolReader::LoadIntoMemory(bool only_in_mem1) const
 
   return true;
 }
+
+// On a real console this would be done in the Espresso bootrom
+bool DolReader::LoadAncastIntoMemory() const
+{
+  // The ancast image will always be in data section 0
+  const auto& section = m_data_sections[0];
+  const u32 section_address = m_dolheader.dataAddress[0];
+
+  const EspressoAncastHeader* header = (const EspressoAncastHeader*)section.data();
+
+  // Verify header block size
+  if (Common::swap32(header->header_block.header_block_size) != sizeof(AncastHeaderBlock))
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid header block size: 0x{:x}", Common::swap32(header->header_block.header_block_size));
+    return false;
+  }
+
+  // Make sure this is a PPC ancast image
+  if (Common::swap32(header->signature_block.signature_type) != 0x01)
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid signature type: 0x{:x}", Common::swap32(header->signature_block.signature_type));
+    return false;
+  }
+
+  // Make sure this is a Wii-Mode ancast image
+  if (Common::swap32(header->info_block.image_type) != ANCAST_IMAGE_TYPE_ESPRESSO_WII)
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid image type: 0x{:x}", Common::swap32(header->info_block.image_type));
+    return false;
+  }
+
+  // Verify the body size
+  const u32 body_size = Common::swap32(header->info_block.body_size);
+  if (body_size + sizeof(EspressoAncastHeader) > section.size())
+  {
+    ERROR_LOG_FMT(BOOT, "Ancast: Invalid body size: 0x{:x}", body_size);
+    return false; 
+  }
+
+  // Verify the body hash
+  const auto digest = Common::SHA1::CalculateDigest(section.data() + sizeof(EspressoAncastHeader), body_size);
+  if (std::memcmp(digest.data(), header->info_block.body_hash, Common::SHA1::DIGEST_LEN) != 0)
```

Comment:
```suggestion
  if (digest != header->info_block.body_hash)
```
(with the std::array change)